staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -122,9 +124,45 @@ func SerializeObject(mediaType string, encoder runtime.Encoder, innerW http.Resp
 	}
 }
 
+type debugResponseWriter struct {
+	delegate             http.ResponseWriter
+	write                int64
+	writeCompleted       int64
+	writeHeader          int64
+	writeHeaderCompleted int64
+}
+
+func (d *debugResponseWriter) Header() http.Header {
+	return d.delegate.Header()
+}
+func (d *debugResponseWriter) Write(data []byte) (int, error) {
+	atomic.AddInt64(&d.write, int64(len(data)))
```

Comment:
We don't seem to use the length information below in the defer section, should we just make it `atomic.AddInt64(&d.write, 1)` like the others? (just in case someone tries to write a empty data array, that would throw us off too)